.github/workflows/ghcr-cleaner.yml

```diff
@@ -0,0 +1,27 @@
+name: Github Container Registry Cleaner
+on:
+  schedule:
+    - cron: "0 4 * * Sun"
+
+jobs:
+  delete_old_images:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Delete untagged images
+        uses: dataaxiom/ghcr-cleanup-action@v1
+        with:
+          dry-run: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Delete older tagged images
+        uses: dataaxiom/ghcr-cleanup-action@v1
+        with:
+          tags: main, latest, v*.*.*
+          keep-n-tagged: 1
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Delete tagged images for pull requests
+        uses: dataaxiom/ghcr-cleanup-action@v1
+        with:
+          tags: pr*
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Yeah good point about snok github action, this is what i tried out first but this does not take into account the multiplatform tags which was the main issue :s

Reply:
I cleaned up the git history but it was in there before

Reply:
> Deleting all (truly) untagged images (that have no reference to any parent) sounds good.
> 
> As we can not use timestamps, for this solution we'd have to use a 'best guess' as what to keep or remove. I suggest keeping 'the last 10 images' (including the multi-builds that depend on it), and always keep everything marked as 'release' or main.. this strategy does require a 'release' branch or another way to mark 'release' images.. as main can (but does not have to be) the release currently in production (again, depending on choices).
> 
> ```
> jobs:
>   - name: clean up repository images
>     runs-on: ubuntu-latest
>     steps:
>       - name: 'Clean up development images'
>         uses: dataaxiom/ghcr-cleanup-action@v1
>         with:
>           keep-n-tagged: 10
>           exclude-tags: '*release*,*main*,*latest*'
>           token: ${{ secrets.GITHUB_TOKEN }}
>       - name: 'Clean up images from main'
>         uses: dataaxiom/ghcr-cleanup-action@v1
>         with:
>           keep-n-tagged: 5
>           tags: '*main*'
>           token: ${{ secrets.GITHUB_TOKEN }}
> ```

Yes i like this will incorporate it :)

Reply:
> Another solution might be to use the 'original' GitHub cleaner (can't find the link right now), which allows us to use filters and timestamps, so we can set '30 days delete dev images'. And combine it with another plugin, like the one used here or https://github.com/pcasteran/ghcr-cleaning-action?tab=readme-ov-file, which would run afterwards to clean up all 'orphan' images not removed by the original plugin. Would be a 'best of both worlds' combination IMO.

I would like this idea, but then if we have an multiplatform release of a year ago then it would still be deleted as it is not tagged by the release. The 'orphan's would already be deleted by the current first step of the code